plugin/trino-mariadb/src/main/java/io/trino/plugin/mariadb/MariaDbClient.java

```diff
@@ -457,7 +459,7 @@ public void renameColumn(ConnectorSession session, JdbcTableHandle handle, JdbcC
             execute(connection, sql);
         }
         catch (TrinoException e) {
-            if (e.getCause() instanceof SQLSyntaxErrorException) {
+            if (e.getCause() instanceof SQLSyntaxErrorException syntaxError && syntaxError.getErrorCode() == PARSE_ERROR) {
```

Comment:
> SQLSyntaxErrorException contains invalid column name.

add a code comment 

```
// Note: SQLSyntaxErrorException can be throw also when column name is invalid
```

BTW do we need this complicated logic?
why not "drop support" for the older MariaDB version and simplify this?

Reply:
I think throwing an MariaDB exception without conversion is also fine. Let me handle in a follow-up PR.